htlcswitch/hop/iterator_test.go

```diff
@@ -143,3 +152,501 @@ func TestForwardingAmountCalc(t *testing.T) {
 		})
 	}
 }
+
+// A simplified implementation of the BlindHopProcessor interface.
+type mockBlindHopProcessor struct{}
+
+// For the sake of testing, we assume that the payload is already decrypted.
+// From the perspective of the link, we expect this function implementation
+// (sphinx, test, or otherwise) to deliver us a proper serialized route blinding
+// payload, which we can then parse into a BlindHopPayload{}.
+func (b *mockBlindHopProcessor) DecryptBlindedData(
+	blindingPoint *btcec.PublicKey, payload []byte) ([]byte, error) {
+
+	return payload, nil
+}
+
+// For simplicity's sake we just pass back the same blinding point.
+func (b *mockBlindHopProcessor) NextEphemeral(
+	blindingPoint *btcec.PublicKey) (*btcec.PublicKey, error) {
+
+	return blindingPoint, nil
+}
+
+type crossPayloadValidationTest struct {
+	name                   string
+	onionPayload           *Payload
+	routeBlindingPayload   *BlindedRouteData
+	updateAddBlindingPoint *btcec.PublicKey
+	incomingAmt            lnwire.MilliSatoshi
+	incomingCltv           uint32
+	isFinalHop             bool
+	expectedErr            error
+}
+
+var crossPayloadValidationTests = []crossPayloadValidationTest{
+	{
+		name: "introduction node blinded route",
+		onionPayload: &Payload{
+			blindingPoint: testPubKey, // intro nodes have blinding point in onion
+		},
+		routeBlindingPayload: &BlindedRouteData{
+			ShortChannelID: &testNextHop,
+			RelayInfo: &lnwire.PaymentRelayInfo{
+				BaseFee:         0,
+				FeeRate:         10,
+				CltvExpiryDelta: 10,
+			},
```

Comment:
TODO: maybe see if we can cut down on the verbosity of having to repeat structs like this again and again in these test cases?

Reply:
You could decalre `BlindedRouteData` in the test and then just add test case vars for the fields that are changing?

Reply:
Yeah, that or since I don't think this `PaymentRelayInfo` struct changes for any test I have, I could just declare it once and reuse. I'll look into it.